api/client/network/connect.go

```diff
@@ -0,0 +1,65 @@
+package network
+
+import (
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/client"
+	"github.com/docker/docker/cli"
+	runconfigopts "github.com/docker/docker/runconfig/opts"
+	"github.com/docker/engine-api/types/network"
+	"github.com/spf13/cobra"
+)
+
+type connectOptions struct {
+	network     string
+	container   string
+	ipaddress   string
+	ipv6address string
+	links       []string
+	aliases     []string
+}
+
+func newConnectCommand(dockerCli *client.DockerCli) *cobra.Command {
+	opts := connectOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "connect [OPTIONS] NETWORK CONTAINER",
+		Short: "Connects a container to a network",
+		Args:  cli.RequiresExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.network = args[0]
+			opts.container = args[1]
+			return runConnect(dockerCli, opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&opts.ipaddress, "ip", "", "IP Address")
+	flags.StringVar(&opts.ipv6address, "ip6", "", "IPv6 Address")
+	flags.StringSliceVar(&opts.links, "links", []string{}, "Add link to another container")
+	flags.StringSliceVar(&opts.aliases, "alias", []string{}, "Add network-scoped alias for the container")
+
+	return cmd
+}
+
+func runConnect(dockerCli *client.DockerCli, opts connectOptions) error {
+	// Validate links
+	for _, link := range opts.links {
+		if _, err := runconfigopts.ValidateLink(link); err != nil {
+			return err
+		}
+	}
```

Comment:
I think it would be better to continue to use `opts.NewListOpts(runconfigopts.ValidateLink)` for this.  It's nice when all the flag/arg validation happens at the same spot, so the error messages are consistent (you get the usage message). You can use it with `flags.Var(&opts.links, ...)`.

With this setup I don't think you'll get the usage info.